haystack_experimental/dataclasses/chat_message.py

```diff
@@ -18,6 +18,32 @@
 
 ChatMessageContentT = Union[TextContent, ToolCall, ToolCallResult, ImageContent]
 
+def _deserialize_content_part(part: Dict[str, Any]) -> ChatMessageContentT:
```

Comment:
`_deserialize_content_part ` and `_serialize_content_part` have been extracted to be used in the Jinja2 extension.